.github/workflows/master.yml

```diff
@@ -9,148 +9,233 @@ on:
     - master
 
 jobs:
+  # TODO(SPARK-32248): Recover JDK 11 builds
+  # Build: build Spark and run the tests for specified modules.
   build:
-
+    name: "Build modules: ${{ matrix.modules }} ${{ matrix.comment }} (JDK ${{ matrix.java }}, ${{ matrix.hadoop }}, ${{ matrix.hive }})"
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
-        java: [ '1.8', '11' ]
-        hadoop: [ 'hadoop-2.7', 'hadoop-3.2' ]
-        hive: [ 'hive-1.2', 'hive-2.3' ]
-        exclude:
-        - java: '11'
-          hive: 'hive-1.2'
-        - hadoop: 'hadoop-3.2'
-          hive: 'hive-1.2'
-    name: Build Spark - JDK${{ matrix.java }}/${{ matrix.hadoop }}/${{ matrix.hive }}
-
+        java:
+          - 1.8
+        hadoop:
+          - hadoop3.2
+        hive:
+          - hive2.3
```

Comment:
hive-1.2 and hadoop-2.7 are also removed only for now?

Reply:
Yeah, I filed a JIRA to store it at SPARK-32255. I was thinking we can periodically run. The current PR mimics our default PR builder behaviour for now.

Reply:
We can change hive and hadoop profile on Jenkins test by changing the PR title. I think it is not possible with Github Actions?

Reply:
Yeah, I think this isn't possible at this moment. It would be great if we can figure out a way to do it.